packages/@aws-cdk/aws-cognito/lib/user-pool.ts

```diff
@@ -477,6 +477,31 @@ export interface DeviceTracking {
   readonly deviceOnlyRememberedOnUserPrompt: boolean;
 }
 
+/**
+ * UserPooolAddOns settings for the user pool.
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cognito-userpool-userpooladdons.html
+ */
+export interface UserPoolAddOnsSettings {
+  /**
+   * The user pool's Advanced Security Mode
+   * @default AdvancedSecurityMode.OFF
+   */
+  readonly advancedSecurityMode?: AdvancedSecurityMode;
```

Comment:
Please move this to the top level

https://github.com/aws/aws-cdk/blob/master/docs/DESIGN_GUIDELINES.md#flat